src/BenchmarkDotNet/Toolchains/MonoWasm/WasmGenerator.cs

```diff
@@ -1,27 +1,82 @@
 ﻿using System.IO;
 using System.Text;
+using BenchmarkDotNet.Environments;
 using BenchmarkDotNet.Extensions;
 using BenchmarkDotNet.Helpers;
 using BenchmarkDotNet.Loggers;
 using BenchmarkDotNet.Running;
 using BenchmarkDotNet.Toolchains.CsProj;
 
+
 namespace BenchmarkDotNet.Toolchains.MonoWasm
 {
     public class WasmGenerator : CsProjGenerator
     {
         private readonly string CustomRuntimePack;
+        //TODO: Set this properly
+        private readonly bool Aot = true;
 
-        public WasmGenerator(string targetFrameworkMoniker, string cliPath, string packagesPath, string customRuntimePack)
+        public WasmGenerator(string targetFrameworkMoniker, string cliPath, string packagesPath, string customRuntimePack, bool aot)
             : base(targetFrameworkMoniker, cliPath, packagesPath, runtimeFrameworkVersion: null)
         {
+            Aot = aot;
             CustomRuntimePack = customRuntimePack;
         }
 
+
         protected override void GenerateProject(BuildPartition buildPartition, ArtifactsPaths artifactsPaths, ILogger logger)
         {
+            WasmRuntime runtime = (WasmRuntime)buildPartition.Runtime;
+
+            if (runtime.Aot)
+            {
+                GenerateProjectAot(buildPartition, artifactsPaths, logger);
+            }
+            else
+            {
+                GenerateProjectInterpreter(buildPartition, artifactsPaths, logger);
+            }
+
+        }
+
+        protected  void GenerateProjectAot(BuildPartition buildPartition, ArtifactsPaths artifactsPaths, ILogger logger)
+        {
+            BenchmarkCase benchmark = buildPartition.RepresentativeBenchmarkCase;
+            var projectFile = GetProjectFilePath(benchmark.Descriptor.Type, logger);
+
+            WasmRuntime runtime = (WasmRuntime)buildPartition.Runtime;
+
+            using (var file = new StreamReader(File.OpenRead(projectFile.FullName)))
+            {
+                var (customProperties, sdkName) = GetSettingsThatNeedsToBeCopied(file, projectFile);
+
+                string content = new StringBuilder(ResourceHelper.LoadTemplate("WasmAotCsProj.txt"))
+                    .Replace("$PLATFORM$", buildPartition.Platform.ToConfig())
+                    .Replace("$CODEFILENAME$", Path.GetFileName(artifactsPaths.ProgramCodePath))
+                    .Replace("$CSPROJPATH$", projectFile.FullName)
+                    .Replace("$TFM$", TargetFrameworkMoniker)
+                    .Replace("$PROGRAMNAME$", artifactsPaths.ProgramName)
+                    .Replace("$RUNTIMESRCDIR$", runtime.RuntimeSrcDir.ToString())
+                    .Replace("$COPIEDSETTINGS$", customProperties)
+                    .Replace("$CONFIGURATIONNAME$", buildPartition.BuildConfiguration)
+                    .Replace("$RUNTIMECONFIG$", "Debug")
+                    .Replace("$SDKNAME$", sdkName)
+                    .Replace("$TRIM$", runtime.Aot ? "true" : "false")
```

Comment:
`runtime.Aot` will always be true for `GenerateProjectAot` method?

```suggestion
                    .Replace("$TRIM$", "true")
```